.github/workflows/ci-build.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+
+on:
+  push:
+    branches:
+    tags:
+  pull_request:
+  
+jobs:
+  build:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        php: [7.1, 7.2, 7.3, 7.4]
+        operating-system: [ubuntu-18.04]
+        env: ['EXECUTOR= DEPENDENCIES=--prefer-lowest', 'EXECUTOR=coroutine DEPENDENCIES=--prefer-lowest', 'EXECUTOR=', 'EXECUTOR=coroutine']
+        include:
+          - os: ubuntu-18.04
+            php: 7.1
+            env: EXECUTOR=coroutine
+    steps:
+    - uses: actions/checkout@v1
+    - name: Install PHP
+      uses: shivammathur/setup-php@1.5.2
+      with:
+        php-version: ${{ matrix.php  }}
+        extension-csv: dom, mbstring
+        coverage: xdebug
+    - name: Get Composer Cache Directory
+      id: composer-cache
+      run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+    - name: Cache dependencies
+      uses: actions/cache@v1
+      with:
+        path: ${{ steps.composer-cache.outputs.dir }}
+        key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.json') }}
+        restore-keys: ${{ runner.os }}-composer-
+    - name: Install dependencies
+      run: |
+        export $ENV
+        composer update ${DEPENDENCIES}
+      env:
+        ENV: ${{ matrix.env }}
+    - name: Run unit tests
+      run: |
+        export $ENV
+        ./vendor/bin/phpunit --group default,ReactPromise
+      env:
+        ENV: ${{ matrix.env}}
+    - name: Code coverage
+      if: matrix.php == 7.1 && matrix.env == 'EXECUTOR=coroutine'
+      run: |
+        export $ENV
+        ./vendor/bin/phpunit --coverage-php /tmp/coverage/clover_executor.cov
+        ./vendor/bin/phpunit --coverage-php /tmp/coverage/clover_executor-coroutine.cov
+        ./vendor/bin/phpcov merge /tmp/coverage --clover /tmp/clover.xml
+        wget https://github.com/scrutinizer-ci/ocular/releases/download/1.5.2/ocular.phar
+        php${{matrix.php}} ocular.phar code-coverage:upload --format=php-clover /tmp/clover.xml
+      env:
+        ENV: ${{ matrix.env }}
+    - name: Coding Standard
+      if: matrix.php == 7.1
+      run: |
+        composer update --prefer-dist
+        composer lint
+    - name: PHPStan
+      if: matrix.php == 7.1
```

Comment:
Currently, PHPStan is run 4 times in all these

![image](https://user-images.githubusercontent.com/327717/69641043-c1f1ac80-105f-11ea-99dd-5b68258938d6.png)

I'd prefer to have only PHPStan job, same for CS and I guess Coverage

Reply:
@simPod, i added only one step for  PHPStan job, CS and Coverage

Reply:
I see but eg. unit tests are still run for CS.

https://github.com/Smolevich/graphql-php/runs/328856568

I believe we have to split it into more jobs. We should have similar structure as is in Travis now. To have a job for 1. running tests, then another job to run 2. codesniffer and another job to run 3. phpstan.

It would also be nice to reuse some parts of the config to keep it dry, this might help https://confluence.atlassian.com/bitbucket/yaml-anchors-960154027.html

Your work is definitely appreciated but I'd love to do this right so even though this works, Travis build is currently still superior. Also another problem is that it's hard to get inspired elsewhere, I don't know of any project that has good Actions setup.

Reply:
For example https://github.com/dunglas/vulcain/blob/master/.github/workflows/ci.yml 
https://github.com/api-platform/api-platform/blob/master/.github/workflows/ci-cd.yml

Reply:
I add if condition for phpunit tests step
I don't understand why you attach link to bitbucket docs, but there is no using anchors in current travis configuration
Github actions doen't support anchors in yaml config yet. Discussing https://github.community/t5/GitHub-Actions/Support-for-YAML-anchors/td-p/30336

Reply:
@simPod can see on PR again?

Reply:
I thought those anchors might help to clear up things that might repeat. If it's not supported, no can do :)